gsdx: cmake: don't display "unused parameters" warnings on release bu…

remeh opened the pull request:
This PR remove all the "unused-parameter" thrown during a release build of the GSdx plugin with Clang.

Tested on Clang 3.7.0

gregory38 commented:
I think it will be better to do it in cmake/BuildParameters.cmake. This way, it will apply to all subproject.

remeh commented:
Yoops, ok I'll change that!

remeh commented:
I wanted to check before affirming it but I think it is the `-W` parameter (the line I've modified) which re-active the `-Wunused-parameter` flags. That's why I've added it here precisely *after* this `-W`.
And after a quick check, the -Wno-unused-parameter is already in `BuildParameters.cmake`

gregory38 commented:
Oh. In this case, it is a good opportunity to clean the warning management. The idea is to put generic flags inside BuildParameters.cmake and only enable specific warning for subpart (ideally it should be empty)
